api/src/routes/challenge.test.ts

```diff
@@ -433,6 +496,259 @@ describe('challengeRoutes', () => {
         });
       });
     });
+
+    describe('/modern-challenge-completed', () => {
+      describe('validation', () => {
+        test('POST rejects requests without ids', async () => {
+          const response = await superRequest('/modern-challenge-completed', {
+            method: 'POST',
+            setCookies
+          });
+
+          expect(response.statusCode).toBe(400);
+          expect(response.body).toStrictEqual(
+            isValidChallengeCompletionErrorMsg
+          );
+        });
+
+        test('POST rejects requests without valid ObjectIDs', async () => {
+          const response = await superRequest('/modern-challenge-completed', {
+            method: 'POST',
+            setCookies
+          }).send({ id: 'not-a-valid-id' });
+
+          expect(response.statusCode).toBe(400);
+          expect(response.body).toStrictEqual(
+            isValidChallengeCompletionErrorMsg
+          );
+        });
+      });
+
+      describe('handling', () => {
+        afterEach(async () => {
+          await fastifyTestInstance.prisma.user.updateMany({
+            where: { email: 'foo@bar.com' },
+            data: {
+              completedChallenges: [],
+              savedChallenges: [],
+              progressTimestamps: []
+            }
+          });
+        });
+
+        // HTML(0), JS(1), Modern(6), Video(11), The Odin Project(15)
+        test('POST accepts challenges without files present', async () => {
+          const now = Date.now();
+
+          const response = await superRequest('/modern-challenge-completed', {
+            method: 'POST',
+            setCookies
+          }).send(HtmlChallengeBody);
+
+          const user = await fastifyTestInstance.prisma.user.findFirst({
```

Comment:
```suggestion
          const user = await fastifyTestInstance.prisma.user.findFirstOrThrow({
```
It allows us to drop some `?`s and surfaces the error earlier making for easier debugging.